peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+PEP: 817
+Title: Wheel Variants: Beyond Platform Tags
+Author: Jonathan Dekhtiar <jonathan@dekhtiar.com>,
+        Michał Górny <mgorny@quansight.com>,
+        Konstantin Schütze <konstin@mailbox.org>,
+        Ralf Gommers <ralf.gommers@gmail.com>,
+        Andrey Talman <atalman@meta.com>,
+        Charlie Marsh <charlie@astral.sh>,
+        Michael Sarahan <msarahan@gmail.com>,
+        Eli Uriegas <eliuriegas@meta.com>,
+        Barry Warsaw <barry@python.org>,
+        Donald Stufft <donald@stufft.io>,
+        Andy R. Terrel <andy.terrel@gmail.com>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 10-Dec-2025
+Post-History:
+
+
+Abstract
+========
+
+The Python wheel packaging format uses
+:doc:`packaging:specifications/platform-compatibility-tags` to specify
+a given wheel's supported environments. These tags are unable to express
+features of modern hardware configurations, such as the availability of
+GPU acceleration, or to provide custom package variants, such as builds
+against different dependency ABIs. This is particularly challenging for
+the scientific computing, artificial intelligence (AI), machine learning
+(ML), and high-performance computing (HPC) communities.
+
+This PEP proposes "Wheel Variants", an extension to the
+:doc:`packaging:specifications/binary-distribution-format`. This
+extension introduces a mechanism for package maintainers to declare
+multiple build variants for the same package version, while allowing
+installers to automatically select the most appropriate variant based on
+system hardware and software characteristics. More specifically, it
+proposes:
+
+- An evolution of the wheel format called **Wheel Variant** that allows
+  wheels to be distinguished by hardware or software attributes.
+
+- A **variant provider plugin** interface that allows installers to
+  dynamically detect platform attributes and select the most suitable
+  wheel.
+
+The goal is for a familiar ``[uv] pip install <package>`` to provide
+the best user experience.
+
+
+Motivation
+==========
+
+The Python packaging ecosystem has evolved to support increasingly
+diverse computing environments. The current software ecosystem often
```

Comment:
"software ecosystem" here is software in general rather than just Python? Would be useful to disambiguate from "packaging ecosystem" in the previous sentence.